tests/gateway/tools.py

```diff
@@ -96,3 +97,20 @@ async def __aenter__(self):
 
     async def __aexit__(self, exc_type, exc, traceback):
         pass
+
+
+def mock_http_client(mock_response: MockAsyncResponse):
+    class MockHttpClient(mock.Mock):
```

Comment:
We can pull this class out from this funciton. IMO, an inner class or function like this makes code difficult to read.

Reply:
Pulled the class out